gsy_e_sdk/setups/Asset_API_scripts/redis_basic_strategies.py

```diff
@@ -0,0 +1,250 @@
+            self.asset_strategy[area_uuid][
+                "fee_to_market_maker"
+            ] = self.calculate_grid_fee(
+                area_uuid,
+                self.get_uuid_from_area_name("Market Maker"),
+                "current_market_fee",
+            )
+
+            # Consumption strategy
+            if "energy_requirement_kWh" in area_dict["asset_info"]:
+                load_strategy = []
+                for i in range(0, TICKS):
+                    if i < TICKS - 2:
+                        load_strategy.append(
+                            round(
+                                fit_rate
+                                - self.asset_strategy[area_uuid]["fee_to_market_maker"]
+                                + (
+                                        market_maker_rate
+                                        + 2
+                                        * self.asset_strategy[area_uuid][
+                                            "fee_to_market_maker"
+                                        ]
+                                        - fit_rate
+                                )
+                                * (i / TICKS),
+                                3,
+                            )
+                        )
+                    else:
+                        load_strategy.append(
+                            round(
+                                market_maker_rate
+                                + self.asset_strategy[area_uuid]["fee_to_market_maker"],
+                                3,
+                            )
+                        )
+                self.asset_strategy[area_uuid]["buy_rates"] = load_strategy
+
+            # Generation strategy
+            if "available_energy_kWh" in area_dict["asset_info"]:
+                gen_strategy = []
+                for i in range(0, TICKS):
+                    if i < TICKS - 2:
+                        gen_strategy.append(
+                            round(
+                                max(
+                                    0,
+                                    market_maker_rate
+                                    + self.asset_strategy[area_uuid][
+                                        "fee_to_market_maker"
+                                    ]
+                                    - (
+                                            market_maker_rate
+                                            + 2
+                                            * self.asset_strategy[area_uuid][
+                                                "fee_to_market_maker"
+                                            ]
+                                            - fit_rate
+                                    )
+                                    * (i / TICKS),
+                                ),
+                                3,
+                            )
+                        )
```

Comment:
Of course, this is personal taste and if you think that the users prefer this way of indention, please ignore me. However, I think it is not readable and propose the following where you separate the calculation of the rate from the rounding and max():
```suggestion
    sell_rate = market_maker_rate + self.asset_strategy[area_uuid]["fee_to_market_maker"] - (
            market_maker_rate + 2 * self.asset_strategy[area_uuid]["fee_to_market_maker"] - fit_rate) * (i / TICKS)

    gen_strategy.append(round(max(0, sell_rate)), 3))
```
Can you please explain why you round the values?

Reply:
Sorry, the indention in the diff is not pretty.
Have a look here:
![Screenshot 2022-06-03 at 14 20 28](https://user-images.githubusercontent.com/19326608/171852476-8e9c2a55-7980-4ba9-8487-4f8c9ce72841.png)

Reply:
The indentation I have proposed is the one Black suggests. But I agree, it is not readable and I will change it. 

Regarding the round function, I investigated, and it had been initially added just to avoid extremely long numbers in the terminal. I believe we can remove the round function and avoid this 'esthetic' issue somewhere else in the backend? What do you think?

Reply:
> the one Black suggests 
Who is "Black"?  

Regarding the rounding: If you are ok with sending the rounded values to the exchange, all good. I thought that this rounding could mask one of the bidding strategies' behaviour.

Reply:
Black is a formatting tool: https://pypi.org/project/black/.

Reply:
I guess this proves that Black is not useful :-)